kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -55,31 +61,40 @@ public RegisteredSchema getSchema(
       Optional<String> rawSchema,
       boolean isKey) {
     // (subject|subjectNameStrategy)?, schemaId
+
+    if (!schemaRegistryEnabled) {
```

Comment:
Instead of injecting `schemaRegistryEnabled`, what if you created a `ThrowingSchemaManager` and move this logic to the module. Then you can remove `schemaRegistryEnabled` altogether.